DataReport/collectBucketMetricsAndUpdateBucketCapacityInfo.js

```diff
@@ -0,0 +1,126 @@
+}
+
+function isValidCapacityValue(capacity) {
+    return (typeof capacity === 'number' && capacity >= 0);
+}
+
+function collectBucketMetricsAndUpdateBucketCapacityInfo(mongoClient, log, callback) {
+    let bucketMetrics;
+    return async.waterfall([
+        next => mongoClient.readCountItems(log, (err, doc) => {
+            if (err) {
+                log.error('an error occurred during readCountItems', {
+                    error: { message: err.message },
+                });
+                return next(err);
+            }
+            if (doc && doc.dataMetrics && doc.dataMetrics.bucket) {
+                bucketMetrics = doc.dataMetrics.bucket;
+            }
+            return next();
+        }),
+        next => mongoClient.getBucketInfos(log, (err, bucketInfos) => {
+            if (err) {
+                log.error('error getting bucket list', {
+                    error: err,
+                });
+                return next(err);
+            }
+            return next(null, bucketInfos.bucketInfos);
+        }),
+        (bucketInfoList, next) => async.eachLimit(
+            bucketInfoList,
+            MaxParallelLimit,
+            (bucket, cb) => {
+                if (!isSOSCapacityInfoEnabled(bucket)) {
+                    return cb();
+                }
+                const bucketName = bucket.getName();
+
+                // get bucket storage used
+                let bucketStorageUsed = -1;
+                if (isValidBucketStorageMetrics(bucketMetrics[bucketName])) {
+                    bucketStorageUsed = bucketMetrics[bucketName].usedCapacity.current
+                            + bucketMetrics[bucketName].usedCapacity.nonCurrent;
+                }
+
+                // read Capacity from bucket._capabilities
+                let { Capacity } = bucket.getCapabilities().VeeamSOSApi.CapacityInfo;
+
+                let Available;
+                if (isValidCapacityValue(Capacity)) { // is Capacity value is valid
+                    // if bucket storage used is valid and bigger than the reported used capacity
+                    if (bucketStorageUsed !== -1 && (Capacity - bucketStorageUsed) >= 0) {
+                        Available = Capacity - bucketStorageUsed;
+                    } else { // if bucket storage used is not retrievable
+                        Available = -1;
+                    }
+                } else { // is Capacity value is not valid
+                    Capacity = -1;
+                    Available = -1;
+                }
+
+                return mongoClient.updateBucketCapacityInfo(bucketName, {
+                    Capacity,
+                    Available,
+                    Used: bucketStorageUsed,
+                }, log, err => {
+                    if (err) {
+                        log.error('an error occurred during put bucket CapacityInfo attributes', {
+                            error: { message: err.message },
```

Comment:
Might be useful to specify in the error message the current bucket, because we may have this error message multiple time, as it's running inside a loop